src/main/scala/org/ua/ilm/datamining/crawlers/youtube/ParallelCrawlerChannels.scala

```diff
@@ -0,0 +1,22 @@
+package org.ua.ilm.datamining.crawlers.youtube
+
+import java.util.concurrent.{Callable, ExecutorService, Executors}
+
+class ParallelCrawlerChannels(key: String, idChannels: Array[String], partChannels: Option[String], numberOfThreads: Int)
+  extends CrawlerChannels(key, idChannels, partChannels) {
+
+  override def getChannelsOfJson: Array[String] = {
+    val poolThreads: ExecutorService = Executors.newFixedThreadPool(numberOfThreads)
+    val arrCallable = getCallable(idChannels)
+
+    val channelsOfJson = for {callable <- arrCallable} yield poolThreads.submit(callable).get()
+    poolThreads.shutdown()
+    channelsOfJson
+  }
+
+  def getCallable(array: Array[String]): Array[Callable[String]] =
```

Comment:
why get callable? Hoe are you supposing the user to understand what this method will do?